Makefile

```diff
@@ -24,6 +29,14 @@ webpack: test_front_end
 webpack_autoreload: npm
 	cd $(JS_PRIV); $(BIN_DIR)/webpack -w -d
 
+nif:
+	gcc -I $(ERTS_INCLUDE_DIR) \
+	-I $(ERL_INTERFACE_INCLUDE_DIR) \
+	-L $(ERL_INTERFACE_LIB_DIR) \
+	-fPIC -shared \
+	-o $(NIF_DIR)/xprof_core_nif_tracer.so \
```

Comment:
Does the .so extension matter? I mean, will it work as expected on non-linux platforms as well?

BTW we can think about employing rebar 
https://www.rebar3.org/v3/docs/building-cc

Reply:
there is also [port compiler](https://www.rebar3.org/v3/docs/using-available-plugins#port-compiler) rebar3 plugin, I don't know which was is preferred way.